### PR TITLE
Atlas keyboard

### PR DESCRIPTION
**Pull request by awkannan**
## Description

Add the Atlas keyboard from CannonKeys and Acheron

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by awkannan** [comment]
TODO:

- [ ] Add 2x2U layout
- [ ] Fix matrix (figure out pin for Row 2)

File: keyboards/cannonkeys/atlas/readme.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# CannonKeys Atlas
+
+Atlas Keyboard
+
+Keyboard Maintainer: [Andrew Kannan](https://github.com/awkannan)  
+Hardware Supported: STM32F072CBT6  
+
+Make example for this keyboard (after setting up your build environment):
+
+    make cannonkeys/Atlas:default
```
Comment (ridingqwerty) [inline keyboards/cannonkeys/atlas/readme.md]:
```suggestion
    make cannonkeys/atlas:default
```

File: keyboards/cannonkeys/atlas/config.h [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+*/
+
+#pragma once
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xCA04
+#define PRODUCT_ID      0xA7A5
+#define DEVICE_VER      0x0001
+/* in python2: list(u"whatever".encode('utf-16-le')) */
+/*   at most 32 characters or the ugly hack in usb_main.c borks */
```
Comment (ridingqwerty) [inline keyboards/cannonkeys/atlas/config.h]: Should this be here?

File: keyboards/cannonkeys/atlas/atlas.h [hunk 3]
```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include "quantum.h"
+
+#define LAYOUT_1x2uC( \
```
Comment (noroadsleft) [inline keyboards/cannonkeys/atlas/atlas.h]: I think eventually there will be Community Layout support for this, but the four-row version will have the same variant, so we should be specific here.

```suggestion
#define LAYOUT_ortho_5x12_1x2uC( \
```

File: keyboards/cannonkeys/atlas/info.json [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+          {"label":"Space", "x":5, "y":4},
+          {"label":"Space", "x":6, "y":4},
+          {"label":"Raise", "x":7, "y":4},
+          {"label":"Left", "x":8, "y":4},
+          {"label":"Down", "x":9, "y":4},
+          {"label":"Up", "x":10, "y":4},
+          {"label":"Right", "x":11, "y":4}
+        ]
+      },
+      "LAYOUT_1x2uC": {
```
Comment (noroadsleft) [inline keyboards/cannonkeys/atlas/info.json]:
```suggestion
      "LAYOUT_ortho_5x12_1x2uC": {
```

File: keyboards/cannonkeys/atlas/readme.md [hunk 5]
```diff
@@ -0,0 +1,12 @@
+# CannonKeys Atlas
+
+Atlas Keyboard
+
+Keyboard Maintainer: [Andrew Kannan](https://github.com/awkannan)
```
Comment (noroadsleft) [inline keyboards/cannonkeys/atlas/readme.md]:
```suggestion
* Keyboard Maintainer: [Andrew Kannan](https://github.com/awkannan)
```

File: keyboards/cannonkeys/atlas/readme.md [hunk 6]
```diff
@@ -0,0 +1,12 @@
+# CannonKeys Atlas
+
+Atlas Keyboard
+
+Keyboard Maintainer: [Andrew Kannan](https://github.com/awkannan)  
+Hardware Supported: STM32F072CBT6
```
Comment (noroadsleft) [inline keyboards/cannonkeys/atlas/readme.md]:
```suggestion
* Hardware Supported: STM32F072CBT6
```

File: keyboards/cannonkeys/atlas/rules.mk [hunk 7]
```diff
@@ -0,0 +1,21 @@
+# MCU Name
+MCU = STM32F072xB
```
Comment (noroadsleft) [inline keyboards/cannonkeys/atlas/rules.mk]: Specifying the variant (`xB`) currently prevents compiling in QMK Configurator.

```suggestion
MCU = STM32F072
```

File: keyboards/cannonkeys/atlas/rules.mk [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+
+# Build Options
+#   comment out to disable the options.
+#
+
+# project specific files
+
```
Comment (noroadsleft) [inline keyboards/cannonkeys/atlas/rules.mk]: Lines 7-9 should be deleted.

File: keyboards/cannonkeys/atlas/rules.mk [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+BOOTMAGIC_ENABLE = lite	# Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes	# Mouse keys
+EXTRAKEY_ENABLE = yes	# Audio control and System control
+CONSOLE_ENABLE = yes	# Console for debug
+COMMAND_ENABLE = yes    # Commands for debug and configuration
+SLEEP_LED_ENABLE = yes  # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes	    # USB Nkey Rollover
+CUSTOM_MATRIX = no # Custom matrix file
```
Comment (noroadsleft) [inline keyboards/cannonkeys/atlas/rules.mk]:
```suggestion
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = yes        # Console for debug
COMMAND_ENABLE = yes        # Commands for debug and configuration
SLEEP_LED_ENABLE = yes      # Breathing sleep LED during USB suspend
NKRO_ENABLE = yes           # USB Nkey Rollover
CUSTOM_MATRIX = no          # Custom matrix file
```

File: keyboards/cannonkeys/atlas/readme.md [hunk 10]
```diff
@@ -0,0 +1,12 @@
+# CannonKeys Atlas
+
+Atlas Keyboard
```
Comment (noroadsleft) [inline keyboards/cannonkeys/atlas/readme.md:3]: Line 3 should have a photo, or be deleted.

**Comment by awkannan** [comment]
@noroadsleft Done.

@zvecr @ridingqwerty can you guys review?

File: keyboards/cannonkeys/atlas/rules.mk [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+MCU = STM32F072
+
+# Build Options
+#   comment out to disable the options.
+BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+SLEEP_LED_ENABLE = yes      # Breathing sleep LED during USB suspend
```
Comment (zvecr) [inline keyboards/cannonkeys/atlas/rules.mk:11]: No caps lock code, so this wont actually do anything
```suggestion
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
```

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**